src/Microsoft.Azure.WebJobs.Host/Bindings/FunctionBindingContext.cs

```diff
@@ -22,11 +23,18 @@ public class FunctionBindingContext
         /// <param name="functionInstanceId">The instance ID of the function being bound to.</param>
         /// <param name="functionCancellationToken">The <see cref="CancellationToken"/> to use.</param>
         /// <param name="trace">The trace writer.</param>
-        public FunctionBindingContext(Guid functionInstanceId, CancellationToken functionCancellationToken, TraceWriter trace)
+        /// <param name="functionDescriptor">Current function being executed. This will influence properties on the <see cref="SystemBindingData.MethodName"/>  </param>
+        public FunctionBindingContext(
+            Guid functionInstanceId, 
```

Comment:
For this overload you can get the Id from the descriptor, so you don't need this separate Guid param. So:

```csharp
public FunctionBindingContext(FunctionDescriptor descriptor, TraceWriter trace, CancellationToken functionCancellationToken)
```